src/configAPIClient.ts

```diff
@@ -91,78 +90,95 @@ export interface IConfigResponse {
+
+    if (planId) {
+        queryParams.push(`plan_id=${planId}`);
+    }
+
+    if (planVersion) {
+        queryParams.push(`plan_version=${planVersion}`);
+    }
+
+    return `${url}?${queryParams.join('&')}`;
+};
+
+export default function ConfigAPIClient(
+    this: IConfigAPIClient,
+    apiKey: string,
+    config: SDKInitConfig,
+    mpInstance: MParticleWebSDK
+): void {
+    const baseUrl = 'https://' + mpInstance._Store.SDKConfig.configUrl;
+    const { isDevelopmentMode } = config;
+    const dataPlan = config.dataPlan as DataPlanConfig;
+    const uploadUrl = buildUrl(baseUrl, apiKey, dataPlan, isDevelopmentMode);
+    const uploader: AsyncUploader = window.fetch
+        ? new FetchUploader(uploadUrl)
+        : new XHRUploader(uploadUrl);
+
+    this.getSDKConfiguration = async (): Promise<IConfigResponse> => {
+        let configResponse: IConfigResponse;
+        const fetchPayload: fetchPayload = {
+            method: 'get',
+            headers: {
+                Accept: 'text/plain;charset=UTF-8',
+                'Content-Type': 'text/plain;charset=UTF-8',
+            },
+            body: null,
+        };
+
         try {
-            const xhrCallback = function() {
-                if (xhr.readyState === 4) {
-                    // when a 200 returns, merge current config with what comes back from config, prioritizing user inputted config
-                    if (xhr.status === 200) {
-                        config = mpInstance._Helpers.extend(
-                            {},
-                            config,
-                            JSON.parse(xhr.responseText)
-                        );
-                        completeSDKInitialization(apiKey, config, mpInstance);
-                        mpInstance.Logger.verbose(
-                            'Successfully received configuration from server'
-                        );
-                    } else {
-                        // if for some reason a 200 doesn't return, then we initialize with the just the passed through config
-                        completeSDKInitialization(apiKey, config, mpInstance);
-                        mpInstance.Logger.verbose(
-                            'Issue with receiving configuration from server, received HTTP Code of ' +
-                                xhr.status
-                        );
-                    }
-                }
-            };
-
-            const xhr = mpInstance._Helpers.createXHR(xhrCallback);
-            url =
-                'https://' +
-                mpInstance._Store.SDKConfig.configUrl +
-                apiKey +
-                '/config?env=';
-            if (config.isDevelopmentMode) {
-                url = url + '1';
-            } else {
-                url = url + '0';
-            }
+            const response = await uploader.upload(fetchPayload);
+            if (response.status === 200) {
+                console.log(
```

Comment:
use mpinstance.logger instead of console.log

Reply:
No need. There's already an mp logger below. This console.log was just for testing.